File: src/subject_service/SubjectTypes.py
from pydantic import BaseModel

from subject_service.TestTypes import TestInstance



class SubjectInfo(BaseModel):
    id: str
    name: str
    description: str
    owner: str  # username
    teachers: list[str]  # only teacher usernames!


class Subject(BaseModel):
    info: SubjectInfo
    student_access_code: str
    teacher_access_code: str
    students: list[str]  # only student usernames!
    test_instances: list[TestInstance]

File: src/frontend/quiz/src/types/SubjectTypes.ts
import type { TestInstance, Test } from "./TestTypes"

export interface SubjectInfo {
    id: string
    name: string
    description: string
    owner: string,
    teachers: Array<string>,
    
}

export interface Subject {
    info: SubjectInfo
    student_access_code: string
    teacher_access_code: string
    students: Array<string>
    test_instances: Array<TestInstance>
}